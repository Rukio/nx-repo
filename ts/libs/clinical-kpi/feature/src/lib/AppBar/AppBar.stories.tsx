import { StoryFn, Meta } from '@storybook/react';
import { AppBar } from './';

export default {
  component: AppBar,
  title: 'AppBar',
} as Meta<typeof AppBar>;

const Template: StoryFn<typeof AppBar> = (args) => <AppBar {...args} />;

export const Primary = Template.bind({});
Primary.args = { stationURL: 'https://qa.*company-data-covered*.com' };
