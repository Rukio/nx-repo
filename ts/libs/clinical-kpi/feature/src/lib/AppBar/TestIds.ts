export const APP_BAR_TEST_IDS = {
  DISPATCH_HEALTH_LOGO_LINK: 'appbar-dispatch-health-logo-link',
  KPI_DASHBOARD_GRAPHIC: 'appbar-kpi-dashboard-graphic',
  STATION_LINK: 'appbar-station-dashboard-link',
  STATION_LINK_LABEL_PREFIX: 'appbar-station-dashboard-link-label-prefix',
};
