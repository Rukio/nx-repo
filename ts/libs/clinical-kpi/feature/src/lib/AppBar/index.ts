export { default as AppBar } from './AppBar';
export { APP_BAR_TEST_IDS } from './TestIds';
