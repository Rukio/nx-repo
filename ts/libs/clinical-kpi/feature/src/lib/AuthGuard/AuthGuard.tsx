import { useAuth0 } from '@auth0/auth0-react';
import { FC, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import {
  setAccessToken,
  setUser,
  selectAuthAccessToken,
} from '@*company-data-covered*/clinical-kpi/data-access';
import { LoadingContainer } from '@*company-data-covered*/clinical-kpi/ui';

export interface AuthGuardProps {
  children: React.ReactNode;
}

const AuthGuard: FC<AuthGuardProps> = ({ children }: AuthGuardProps) => {
  const {
    user,
    isAuthenticated,
    loginWithRedirect,
    isLoading,
    getAccessTokenSilently,
  } = useAuth0();

  const accessToken = useSelector(selectAuthAccessToken);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(setUser({ user }));
  }, [user, dispatch]);

  useEffect(() => {
    if ((!isAuthenticated || !user) && !isLoading) {
      loginWithRedirect();
    }
  }, [isAuthenticated, user, isLoading, loginWithRedirect]);

  useEffect(() => {
    const getAuthToken = async () => {
      try {
        const accessToken = await getAccessTokenSilently();
        dispatch(setAccessToken({ accessToken: accessToken ?? '' }));
      } catch {
        loginWithRedirect();
      }
    };
    if (!isLoading && isAuthenticated) {
      getAuthToken();
    }
  }, [
    dispatch,
    getAccessTokenSilently,
    isAuthenticated,
    isLoading,
    loginWithRedirect,
  ]);

  return accessToken ? (
    <>{children}</>
  ) : (
    <LoadingContainer testIdPrefix="authguard" />
  );
};

export default AuthGuard;
