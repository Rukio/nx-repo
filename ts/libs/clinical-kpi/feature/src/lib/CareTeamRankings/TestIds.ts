export const CARE_TEAM_RANKINGS_TEST_IDS = {
  SECTION_PREFIX: 'care-team-rankings-section',
  PAGINATION: 'care-team-rankings-pagination',
};
