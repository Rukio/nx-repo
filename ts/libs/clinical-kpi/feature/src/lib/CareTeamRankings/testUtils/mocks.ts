import {
  AuthenticatedUser,
  MarketMetrics,
  mockedAuthenticatedUser,
} from '@*company-data-covered*/clinical-kpi/data-access';
import {
  buildProvider,
  generateIncrementalMockedProviderMetrics,
} from '../../util/testUtils/builders/providerProfile';

export const mockedProviderMetricsResponse: MarketMetrics = {
  providerMetrics: [
    {
      provider: buildProvider(mockedAuthenticatedUser.id),
      metrics: generateIncrementalMockedProviderMetrics(19),
    },
  ],
};

export const mockAuthenticatedUserId: AuthenticatedUser['id'] =
  mockedProviderMetricsResponse.providerMetrics[0].provider.id.toString();
