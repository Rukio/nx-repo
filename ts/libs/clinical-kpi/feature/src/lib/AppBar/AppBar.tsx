import { FC, useMemo } from 'react';
import { NavLink } from 'react-router-dom';
import {
  ArrowForwardIcon,
  Box,
  Button,
  Container,
  Link,
  Typography,
  makeSxStyles,
  SxStylesValue,
} from '@*company-data-covered*/design-system';
import { LEADS_VIEW_INDIVIDUAL_VISIBILITY } from '@*company-data-covered*/clinical-kpi/ui';
import { ReactComponent as *company-data-covered*Logo } from './assets/*company-data-covered*-logo.svg';
import { ReactComponent as KPIDashboardGraphic } from './assets/kpidashboard-graphic.svg';
import { APP_BAR_TEST_IDS } from './TestIds';
import { checkFeatureGate } from '../util/statsigUtils';
import { useUserMarketRolePermission } from '../hooks/useUserMarketRolePermission';
import { ALLOWED_ROLES } from '../constants';

type Props = {
  stationURL: string;
};

type LinkStyles = {
  activeLink: SxStylesValue;
  link: SxStylesValue;
  isActive?: boolean;
};

const makeStyles = () =>
  makeSxStyles({
    container: (theme) => ({
      position: 'relative',
      display: 'flex',
      justifyContent: 'space-between',
      paddingTop: theme.spacing(4),
    }),
    nav: (theme) => ({
      height: '1rem',
      width: '100%',
      lineHeight: '1rem',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-start',
      [theme.breakpoints.down('sm')]: {
        justifyContent: 'space-between',
      },
    }),
    toDashboardButton: (theme) => ({
      ml: theme.spacing(5),
      textTransform: 'none',
      color: theme.palette.primary.main,
    }),
    toDashboardButtonLabelPrefix: (theme) => ({
      [theme.breakpoints.down('sm')]: {
        display: 'none',
      },
    }),
    kpiDashboardGraphic: (theme) => ({
      position: 'absolute',
      right: 0,
      top: 0,
      [theme.breakpoints.down('sm')]: {
        display: 'none',
      },
    }),
    navigationContainer: {
      display: 'flex',
      gap: 2.5,
      marginLeft: '48px',
      fontSize: '14px',
      fontStyle: 'normal',
      fontWeight: '600',
    },
    textDecoration: {
      '& a': {
        textDecoration: 'none',
      },
    },
    activeLink: {
      color: 'black',
    },
    link: {
      color: '#9AA3A9',
    },
  });

const navigationLinks = [
  {
    link: '/',
    label: 'My Performance Hub',
  },
  {
    link: '/leads',
    label: 'Leader Hub',
  },
];

const getLinkStyles = ({ isActive, activeLink, link }: LinkStyles) =>
  isActive ? activeLink : link;

const AppBar: FC<Props> = ({ stationURL }) => {
  const styles = makeStyles();

  const { activeLink, link } = styles;

  const showLeadsViewIndividualVisibility =
    useUserMarketRolePermission(ALLOWED_ROLES);

  const isLeadsViewIndividualVisibility = useMemo(
    () => checkFeatureGate(LEADS_VIEW_INDIVIDUAL_VISIBILITY),
    []
  );

  return (
    <Container sx={styles.container}>
      <Box sx={styles.nav} component="nav">
        <Link
          href={stationURL}
          data-testid={APP_BAR_TEST_IDS.DISPATCH_HEALTH_LOGO_LINK}
        >
          <*company-data-covered*Logo />
        </Link>
        {isLeadsViewIndividualVisibility && showLeadsViewIndividualVisibility && (
          <Box sx={styles.navigationContainer}>
            {navigationLinks.map((navLink, index) => (
              <Box sx={styles.textDecoration}>
                <NavLink key={index} to={navLink.link}>
                  {({ isActive }) => (
                    <Box
                      component="span"
                      sx={{
                        ...getLinkStyles({ isActive, activeLink, link }),
                      }}
                    >
                      {navLink.label}
                    </Box>
                  )}
                </NavLink>
              </Box>
            ))}
          </Box>
        )}
        {!isLeadsViewIndividualVisibility && (
          <Button
            href={stationURL}
            variant="text"
            size="small"
            sx={styles.toDashboardButton}
            endIcon={<ArrowForwardIcon />}
            data-testid={APP_BAR_TEST_IDS.STATION_LINK}
          >
            <Typography
              data-testid={APP_BAR_TEST_IDS.STATION_LINK_LABEL_PREFIX}
              sx={styles.toDashboardButtonLabelPrefix}
            >
              Continue to&nbsp; Dashboard
            </Typography>
          </Button>
        )}
        <Box sx={styles.kpiDashboardGraphic}>
          <KPIDashboardGraphic
            data-testid={APP_BAR_TEST_IDS.KPI_DASHBOARD_GRAPHIC}
          />
        </Box>
      </Box>
    </Container>
  );
};

export default AppBar;
