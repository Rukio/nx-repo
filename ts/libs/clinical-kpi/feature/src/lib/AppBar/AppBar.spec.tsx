import { mocked } from 'jest-mock';
import { useAuth0 } from '@auth0/auth0-react';

import { renderWithUserProvider, screen } from '../util/testUtils';
import AppBar from './AppBar';
import { APP_BAR_TEST_IDS } from './TestIds';
import AuthGuard from '../AuthGuard/AuthGuard';
import { mockedUseAuth0DefaultReturn, user } from '../AuthGuard/AuthGuard.spec';
import { MOCK_JWT_TOKEN_APP, MOCK_JWT_TOKEN_MARKET_MANAGER } from '../mocks';

const mockCheckGate = jest.fn();

jest.mock('statsig-js', () => ({
  checkGate: () => mockCheckGate(),
}));

jest.mock('@auth0/auth0-react');

const mockedUseAuth0 = mocked(useAuth0);

const setup = () => {
  return renderWithUserProvider(
    <AuthGuard>
      <AppBar stationURL="https://qa.*company-data-covered*.com/" />
    </AuthGuard>,
    {}
  );
};

const testRender = async () => {
  const { asFragment } = setup();
  const appBar = await screen.findByTestId(
    APP_BAR_TEST_IDS.DISPATCH_HEALTH_LOGO_LINK
  );
  expect(appBar).toBeVisible();
  // verified in cypress - url values, logo link, continue link, graphic displayed
  expect(asFragment()).toMatchSnapshot();
};

describe('AppBar', () => {
  describe('with allowed roles in JWT', () => {
    beforeEach(() => {
      mockedUseAuth0.mockReturnValue({
        ...mockedUseAuth0DefaultReturn,
        isAuthenticated: true,
        user,
        isLoading: false,
        getAccessTokenSilently: jest
          .fn()
          .mockResolvedValue(MOCK_JWT_TOKEN_MARKET_MANAGER),
      });
    });

    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should render correctly when flag is turned on', async () => {
      mockCheckGate.mockReturnValue(true);
      await testRender();
    });

    it('should render correctly when flag is turned off', async () => {
      mockCheckGate.mockReturnValue(false);
      await testRender();
    });
  });

  describe('without allowed roles in JWT', () => {
    beforeEach(() => {
      mockedUseAuth0.mockReturnValue({
        ...mockedUseAuth0DefaultReturn,
        isAuthenticated: true,
        user,
        isLoading: false,
        getAccessTokenSilently: jest.fn().mockResolvedValue(MOCK_JWT_TOKEN_APP),
      });
    });

    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should render correctly when flag is turned on', async () => {
      mockCheckGate.mockReturnValue(true);
      await testRender();
    });

    it('should render correctly when flag is turned off', async () => {
      mockCheckGate.mockReturnValue(false);
      await testRender();
    });
  });
});
