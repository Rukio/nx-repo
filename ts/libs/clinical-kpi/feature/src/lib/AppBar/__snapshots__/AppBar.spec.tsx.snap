// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar with allowed roles in JWT should render correctly when flag is turned off 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 32px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  height: 1rem;
  width: 100%;
  line-height: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

@media (max-width:599.95px) {
  .emotion-1 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

.emotion-2 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 4px 5px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
  margin-left: 40px;
  text-transform: none;
  color: #1976d2;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

@media (max-width:599.95px) {
  .emotion-4 {
    display: none;
  }
}

.emotion-5 {
  display: inherit;
  margin-right: -2px;
  margin-left: 8px;
}

.emotion-5>*:nth-of-type(1) {
  font-size: 18px;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  position: absolute;
  right: 0;
  top: 0;
}

@media (max-width:599.95px) {
  .emotion-8 {
    display: none;
  }
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <nav
      class="MuiBox-root emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
        data-testid="appbar-dispatch-health-logo-link"
        href="https://qa.*company-data-covered*.com/"
      >
        <svg>
          *company-data-covered*-logo.svg
        </svg>
      </a>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall emotion-3"
        data-testid="appbar-station-dashboard-link"
        href="https://qa.*company-data-covered*.com/"
        tabindex="0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-4"
          data-testid="appbar-station-dashboard-link-label-prefix"
        >
          Continue toÂ  Dashboard
        </p>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeSmall emotion-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root emotion-7"
        />
      </a>
      <div
        class="MuiBox-root emotion-8"
      >
        <svg
          data-testid="appbar-kpi-dashboard-graphic"
        >
          kpidashboard-graphic.svg
        </svg>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`AppBar with allowed roles in JWT should render correctly when flag is turned on 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 32px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  height: 1rem;
  width: 100%;
  line-height: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

@media (max-width:599.95px) {
  .emotion-1 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

.emotion-2 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 20px;
  margin-left: 48px;
  font-size: 14px;
  font-style: normal;
  font-weight: 600;
}

.emotion-4 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  color: black;
}

.emotion-7 {
  color: #9AA3A9;
}

.emotion-8 {
  position: absolute;
  right: 0;
  top: 0;
}

@media (max-width:599.95px) {
  .emotion-8 {
    display: none;
  }
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <nav
      class="MuiBox-root emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
        data-testid="appbar-dispatch-health-logo-link"
        href="https://qa.*company-data-covered*.com/"
      >
        <svg>
          *company-data-covered*-logo.svg
        </svg>
      </a>
      <div
        class="MuiBox-root emotion-3"
      >
        <div
          class="MuiBox-root emotion-4"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <span
              class="MuiBox-root emotion-5"
            >
              My Performance Hub
            </span>
          </a>
        </div>
        <div
          class="MuiBox-root emotion-4"
        >
          <a
            class=""
            href="/leads"
          >
            <span
              class="MuiBox-root emotion-7"
            >
              Leader Hub
            </span>
          </a>
        </div>
      </div>
      <div
        class="MuiBox-root emotion-8"
      >
        <svg
          data-testid="appbar-kpi-dashboard-graphic"
        >
          kpidashboard-graphic.svg
        </svg>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`AppBar without allowed roles in JWT should render correctly when flag is turned off 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 32px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  height: 1rem;
  width: 100%;
  line-height: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

@media (max-width:599.95px) {
  .emotion-1 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

.emotion-2 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 4px 5px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
  margin-left: 40px;
  text-transform: none;
  color: #1976d2;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

@media (max-width:599.95px) {
  .emotion-4 {
    display: none;
  }
}

.emotion-5 {
  display: inherit;
  margin-right: -2px;
  margin-left: 8px;
}

.emotion-5>*:nth-of-type(1) {
  font-size: 18px;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  position: absolute;
  right: 0;
  top: 0;
}

@media (max-width:599.95px) {
  .emotion-8 {
    display: none;
  }
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <nav
      class="MuiBox-root emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
        data-testid="appbar-dispatch-health-logo-link"
        href="https://qa.*company-data-covered*.com/"
      >
        <svg>
          *company-data-covered*-logo.svg
        </svg>
      </a>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall emotion-3"
        data-testid="appbar-station-dashboard-link"
        href="https://qa.*company-data-covered*.com/"
        tabindex="0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-4"
          data-testid="appbar-station-dashboard-link-label-prefix"
        >
          Continue toÂ  Dashboard
        </p>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeSmall emotion-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root emotion-7"
        />
      </a>
      <div
        class="MuiBox-root emotion-8"
      >
        <svg
          data-testid="appbar-kpi-dashboard-graphic"
        >
          kpidashboard-graphic.svg
        </svg>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`AppBar without allowed roles in JWT should render correctly when flag is turned on 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 32px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  height: 1rem;
  width: 100%;
  line-height: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

@media (max-width:599.95px) {
  .emotion-1 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

.emotion-2 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

.emotion-3 {
  position: absolute;
  right: 0;
  top: 0;
}

@media (max-width:599.95px) {
  .emotion-3 {
    display: none;
  }
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <nav
      class="MuiBox-root emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-2"
        data-testid="appbar-dispatch-health-logo-link"
        href="https://qa.*company-data-covered*.com/"
      >
        <svg>
          *company-data-covered*-logo.svg
        </svg>
      </a>
      <div
        class="MuiBox-root emotion-3"
      >
        <svg
          data-testid="appbar-kpi-dashboard-graphic"
        >
          kpidashboard-graphic.svg
        </svg>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
